datalad/downloaders/providers.py

```diff
@@ -337,6 +367,119 @@ def get_provider(self, url, only_nondefault=False):
                   scheme, provider)
         return provider
 
+    def enter_new(self, url=None, auth_types=[]):
+        from datalad.ui import ui
+        name = None
+        if url:
+            ri = RI(url)
+            for f in ('hostname', 'name'):
+                try:
+                    # might need sanitarization
+                    name = str(getattr(ri, f))
+                except AttributeError:
+                    pass
+        known_providers_by_name = {p.name: p for p in self._providers}
+        PROVIDERS_USER_DIR = self._get_providers_dirs()['user']
+        while True:
+            name = ui.question(
+                title="New provider name",
+                text="Unique name to identify 'provider' for %s" % url,
+                default=name
+            )
+            filename = pathjoin(PROVIDERS_USER_DIR, '%s.cfg' % name)
+            if name in known_providers_by_name:
+                if ui.yesno(
+                    title="Known provider %s" % name,
+                    text="Provider with name %s already known. Do you want to "
+                         "use it for this session? If 'No' - enter another name"
```

Comment:
This could be misinterpreted as "To answer 'NO', enter another name here".

Reply:
indeed...  I better move this trailing sentence to avoid confusion - I think it will be ok without it